src/test/groovy/transform/stc/ArraysAndCollectionsSTCTest.groovy

```diff
@@ -27,26 +27,55 @@ class ArraysAndCollectionsSTCTest extends StaticTypeCheckingTestCase {
         assertScript '''
             String[] strings = ['a','b','c']
             String str = strings[0]
+            assert str == 'a'
         '''
     }
 
-    void testArrayElementReturnType() {
+    void testArrayElementTypeInference() {
         shouldFailWithMessages '''
             String[] strings = ['a','b','c']
             int str = strings[0]
         ''', 'Cannot assign value of type java.lang.String to variable of type int'
     }
 
+    void testRightComponentTypeInArray() {
```

Comment:
Minor, but I'd suggest "Compatible" or "Correct" instead of "Right". "Right" is the opposite of "Wrong" but also "Left".